### PR TITLE
SWORD2: Add detection entry for Hebrew V2

### PR DESCRIPTION
**Pull request by orgads**
+ add more files to the current entry, so it won't get dismissed by 3-files match for English.

### CODE REVIEW

File: engines/sword2/detection_tables.h [hunk 1]
```diff
@@ -110,7 +95,7 @@ static const Sword2GameDescription gameDescriptions[] = {
 	{
 		{
 			"sword2",
-			"EU",
+			"Speech",
 			AD_ENTRY3s("general.clu", "31db8564f9187538f24d9fda0677f666", 7059728,
```
Comment (lotharsm) [inline engines/sword2/detection_tables.h:99]: Why was the "EU" flag removed? This indicates a different release (EU vs. US)

**Review by lotharsm — CHANGES_REQUESTED** [review]

**Comment by lotharsm** [comment]
Adding the new version is fine, but please don't modify the existing ones.

**Comment by orgads** [comment]
I removed it because I thought it was a dup of EU, and since both this version and the Hebrew version have 3 files, I always got a prompt to choose. Can you suggest an alternative?

**Comment by lotharsm** [comment]
> I removed it because I thought it was a dup of EU, and since both this 
> version and the Hebrew version have 3 files, I always got a prompt to 
> choose. Can you suggest an alternative?
> 

What you could do is try to check for an INTRO.SMK file for your release 
- with a bit of luck, this one will be unique. I had to get a bit 
creative on the various English releases as well...

**Comment by sev-** [comment]
I do not understand this PR.

It says "Add detection entry for Hebrew V2", but in reality, it:
  - Removed one US release
  - Plays with one EU release description
  - Alternates existing Hebrew version (which is OK)
  - Adds another Hebrew version (as described)

So, what is this PR _really_ about?

File: engines/sword2/detection_tables.h (hunk 1 above)
Comment (orgads) [reply engines/sword2/detection_tables.h]: Recovered

**Comment by orgads** [comment]
> I do not understand this PR.
> 
> It says "Add detection entry for Hebrew V2", but in reality, it:
> 
> * Removes one US release
> * Plays with one EU release description
> * Alternates existing Hebrew version (which is OK)
> * Adds another Hebrew version (as described)
> 
> So, what is this PR _really_ about?

There were several issues with the detection. After a long discussion with @lotharsm we agreed to list 4 files, and now all the versions seem to be correctly detected.

**Comment by lotharsm** [comment]
Thank you, merging!

**Pull request merged**